ReactWindows/ReactNative.Shared/UIManager/UIManagerModule.cs

```diff
@@ -163,12 +178,32 @@ public int AddMeasuredRootView(ReactRootView rootView)
         /// Detaches a root view from the size monitoring hooks in preparation for the unmount
         /// </summary>
         /// <param name="rootView">The root view instance.</param>
-        public void DetachRootView(ReactRootView rootView)
+        public async Task DetachRootView(ReactRootView rootView)
```

Comment:
>DetachRootView [](start = 26, length = 14)

`DetachRootViewAsync`

Reply:
Feels asymmetrical that AddMeasuredRootView is not also async now.

---
In reply to: [177228064](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r177228064) [](ancestors = 177228064)

Reply:
@rozele I noticed I ended up in this state. There are various reasons:
- for various historical reasons apps call the fire and forget StartReactApplication anyway.
- almost all the time the new root view is not measured, so the original StartReactApplicationAsync is cut short and the subsequent initialization is deferred to the measuring callback
- we didn't need any specific steps to keep anything (dispatcher, etc.) alive during initialization. 

Now, it's feasible to make AddMeasuredRootView async and make sure I await for it AFTER we call appRegistry.runApplication, so we don't impact start time.
But the question is, how far/deeper should we go? First level is a  DispatcherHelpers.RunOnDispatcher thing that does stuff that can hardly fail. Then we enqueue a root view creation on LM thread, etc. I think it would be overkill to track the root view completion up to native hierarchy and with no real benefit..




---
In reply to: [177228164](https://github.com/Microsoft/react-native-windows/pull/1737#discussion_r177228164) [](ancestors = 177228164,177228064)